### PR TITLE
Fix the PRESET_INFO test bug

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

**Comment by peternewman** [comment]
Other tests checked for and fixed.

File: python/ola/PidStore.py [hunk 1]
```diff
@@ -116,6 +116,24 @@ def RequestSupported(self, command_class):
     """Check if this PID allows a command class."""
     return self._requests.get(command_class) is not None
 
+  def GetRequest(self, command_class):
+    return self._requests.get(command_class)
+
+  def GetRequestField(self, command_class, field_name):
+    return filter(lambda field: field.name == field_name,
```
Comment (peternewman) [inline python/ola/PidStore.py]: Fixed. We now get a stacktrace just within the tests/test definitions and an error "AttributeError: 'NoneType' object has no attribute 'RawValue'". But it feels kind of redundant to test for it, as it implies either the tests are wrong or the PIDs have changed, either way we'll want to throw an error, so the smaller stack trace covers that.

File: python/ola/PidStore.py [hunk 2]
```diff
@@ -116,6 +116,26 @@ def RequestSupported(self, command_class):
     """Check if this PID allows a command class."""
     return self._requests.get(command_class) is not None
 
+  def GetRequest(self, command_class):
+    return self._requests.get(command_class)
+
+  def GetRequestField(self, command_class, field_name):
+    field = filter(lambda field: field.name == field_name,
+                   self._requests.get(command_class).GetAtoms())
+    return field[0] if (len(field) > 0) else None
```
Comment (nomis52) [inline python/ola/PidStore.py]: I'd change the variable to fields and do

return fields[0] if not fields else None

File: python/ola/PidStore.py [hunk 3]
```diff
@@ -116,6 +116,26 @@ def RequestSupported(self, command_class):
     """Check if this PID allows a command class."""
     return self._requests.get(command_class) is not None
 
+  def GetRequest(self, command_class):
+    return self._requests.get(command_class)
+
+  def GetRequestField(self, command_class, field_name):
+    field = filter(lambda field: field.name == field_name,
+                   self._requests.get(command_class).GetAtoms())
```
Comment (nomis52) [inline python/ola/PidStore.py]: I'd use self.GetRequest(command_class).GetAtoms()

here

File: tools/rdm/TestDefinitions.py [hunk 4]
```diff
@@ -4384,14 +4385,16 @@ def Test(self):
       return
 
     delay_time = self.max_delay_time
+    delay_time_field = self.pid.GetRequestField(RDM_SET, 'loss_of_signal_delay')
     # 0xffff means 'fail mode not supported'
-    if self.max_delay_time * 10 < 0xfffe:
-      delay_time = (self.max_delay_time * 10 + 1) / 10.0  # increment by 1
+    if delay_time_field.RawValue(self.max_delay_time) < (0xffff - 1):
+      delay_time = delay_time_field.DisplayValue(delay_time_field.RawValue(self.max_delay_time) + 1)  # increment by 1
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: Can you wrap at 80 characters please.

File: tools/rdm/TestDefinitions.py [hunk 5]
```diff
@@ -4384,14 +4385,16 @@ def Test(self):
       return
 
     delay_time = self.max_delay_time
+    delay_time_field = self.pid.GetRequestField(RDM_SET, 'loss_of_signal_delay')
     # 0xffff means 'fail mode not supported'
-    if self.max_delay_time * 10 < 0xfffe:
-      delay_time = (self.max_delay_time * 10 + 1) / 10.0  # increment by 1
+    if delay_time_field.RawValue(self.max_delay_time) < (0xffff - 1):
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: Drop the extra parens on the RHS . Parens in not Python style

File: tools/rdm/TestDefinitions.py [hunk 6]
```diff
@@ -4384,14 +4385,16 @@ def Test(self):
...
+      delay_time = delay_time_field.DisplayValue(delay_time_field.RawValue(self.max_delay_time) + 1)  # increment by 1
 
     hold_time = self.max_hold_time
+    hold_time_field = self.pid.GetRequestField(RDM_SET, 'hold_time')
     # 0xffff means 'fail mode not supported'
-    if self.max_hold_time * 10 < 0xfffe:
-      hold_time = (self.max_hold_time * 10 + 1) / 10.0  # increment by 1
+    if hold_time_field.RawValue(self.max_hold_time) < (0xffff - 1):
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: Sam here

**Comment by nomis52** [comment]
Minor comments.

File: python/ola/PidStore.py (hunk 2 above)
Comment (peternewman) [inline python/ola/PidStore.py]: Done.

File: python/ola/PidStore.py (hunk 3 above)
Comment (peternewman) [inline python/ola/PidStore.py]: Done.

File: tools/rdm/TestDefinitions.py (hunk 4 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done.

File: tools/rdm/TestDefinitions.py (hunk 5 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done.

File: tools/rdm/TestDefinitions.py (hunk 6 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done.

**Comment by peternewman** [comment]
This was raised here: https://groups.google.com/forum/#!topic/open-lighting/w7FhOU49_3k

**Pull request merged**